static/js/tools/choropleth/choropleth_template.tsx

```diff
@@ -32,13 +39,132 @@ window.onload = () => {
 };
 
 class MainPane extends Component {
+  constructor(props: Record<string, unknown>) {
+    super(props);
+
+    // Redirect to basic url if none provided.
+    if (window.location.search === "") {
+      window.location.search =
+        "statVar=Count_Person_Employed&pc=t&geoDcid=country/USA";
+    }
+
+    // Bind functions as needed.
+    this._handleStatVarSelection = this._handleStatVarSelection.bind(this);
+
+    // Get default values for optional fields.
+    var urlParams = new URLSearchParams(window.location.search);
+    var isPerCapita = false
+    if (urlParams.has("pc")) {
+      isPerCapita = ["true", "t", "1"].includes(
+        urlParams.get("pc").toLowerCase());
+    }
+
+    // Get all statistical variable available for the current subgeo.
+    axios.get("/api/place/child/statvars/"
+             + urlParams.get("geoDcid")).then((resp) => {
+      let statVars : Set<string> = new Set();
+      // TODO(iancostello): Don't assume the property exists.
+      resp.data.forEach(item => statVars.add(item));
+      // this.setState({
+      //   statsVarFilter: new TimelineStatsVarFilter(statVars)
+      // });
+    });
+
+    // Initialize state.
+    this.state = {
+      // References used for both choropleth and stats var sidemenu objects.
+      choroplethMap: React.createRef(),
+      statVarMenuRef: React.createRef(),
+      pc: isPerCapita,
+      // Default to no filtering in stats var sidemenu until API call returns.
+      statsVarFilter: new NoopStatsVarFilter(), 
+      // Tracks the currently selected node in sidemenu.
+      statsVarNodes: {},
+    };
+  }
+
+  /**
+   * Toggles the per capita value of the choropleth map and redraws the map.
+   */
+  _togglePerCapita() {
+    // Update locally in parent.
+    var newPerCapitaValue = !this.state['pc'];
+    this.setState({
+      pc: newPerCapitaValue
+    })
+    // Redraw the choropleth map.
+    var choroplethRef = this.state['choroplethMap'].current;
+    if (choroplethRef) {
```

Comment:
What should happen if the reference does not exist?  What does this mean?

Reply:
It's checking to make sure that the child is rendered. If not then the reference.current would be null

Reply:
It's really unneeded so I will remove it